src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -889,7 +1285,7 @@ export class SavedObjectsRepository {
     const raw = this._serializer.savedObjectToRaw(migrated as SavedObjectSanitizedDoc);
 
     const response = await this._writeToCluster('update', {
-      id: this._serializer.generateRawId(namespace, type, id),
+      id: raw._id,
```

Comment:
shouldn't we use `generateRawId(namespace, type, id)` here?

Reply:
`savedObjectToRaw` uses this for the `_id` field in the object that it returns. I think the way it is written is a bit clearer because we are doing the update on the raw object.

Reply:
The raw object's id is already converted, so I think this is fine?

Reply:
Ah yes, I missed that, makes sense.